pkg/appgateserver/cmd/cmd.go

```diff
@@ -0,0 +1,186 @@
+	cmd := &cobra.Command{
+		Use:   "appgate-server",
+		Short: "Starts the AppGate server",
+		Long: `Starts the AppGate server that listens for incoming relay requests and handles
+the necessary on-chain interactions (sessions, suppliers, etc) to receive the
+respective relay response.
+
+-- App Mode (Flag)- -
+If the server is started with a defined '--self-signing' flag, it will behave
+as an Application. Any incoming requests will be signed by using the private
+key and ring associated with the '--signing-key' flag.
+
+-- Gateway Mode (Flag)--
+If the '--self-signing' flag is not provided, the server will behave as a Gateway.
+It will sign relays on behalf of any Application sending it relays, provided
+that the address associated with '--signing-key' has been delegated to. This is
+necessary for the application<->gateway ring signature to function.
+
+-- App Mode (HTTP) --
+If an application doesn't provide the '--self-signing' flag, it can still send
+relays to the AppGate server and function as an Application, provided that:
+1. Each request contains the '?senderAddress=[address]' query parameter
+2. The key associated with the '--signing-key' flag belongs to the address
+   provided in the request, otherwise the ring signature will not be valid.`,
+		Args: cobra.NoArgs,
+		RunE: runAppGateServer,
+	}
+
+	cmd.Flags().StringVar(&flagSigningKey, "signing-key", "", "The name of the key that will be used to sign relays")
+	cmd.Flags().StringVar(&flagListeningEndpoint, "listening-endpoint", "http://localhost:42069", "The host and port that the appgate server will listen on")
+	cmd.Flags().StringVar(&flagCometWebsocketUrl, "comet-websocket-url", "ws://localhost:36657/websocket", "The URL of the comet websocket endpoint to communicate with the pocket blockchain")
+	cmd.Flags().BoolVar(&flagSelfSigning, "self-signing", false, "Whether the server should sign all incoming requests with its own ring (for applications)")
+
+	cmd.Flags().String(flags.FlagKeyringBackend, "", "Select keyring's backend (os|file|kwallet|pass|test)")
+	cmd.Flags().String(flags.FlagNode, "tcp://localhost:36657", "The URL of the comet tcp endpoint to communicate with the pocket blockchain")
+
+	return cmd
+}
+
+func runAppGateServer(cmd *cobra.Command, _ []string) error {
+	// Create a context that is canceled when the command is interrupted
+	ctx, cancelCtx := context.WithCancel(cmd.Context())
+	defer cancelCtx()
+
+	// Retrieve the client context for the chain interactions.
+	clientCtx := cosmosclient.GetClientContextFromCmd(cmd)
+
+	// Parse the listening endpoint.
+	listeningUrl, err := url.Parse(flagListeningEndpoint)
+	if err != nil {
+		return fmt.Errorf("failed to parse listening endpoint: %w", err)
+	}
+
+	log.Printf("INFO: Creating block client, using comet websocket URL: %s...", flagCometWebsocketUrl)
+
+	// Create the block client with its dependency on the events client.
+	eventsQueryClient := eventsquery.NewEventsQueryClient(flagCometWebsocketUrl)
```

Comment:
We should follow the dependency construction used in PR #177 [1] and have a function that is responsible for that